doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,145 @@
+# SONiC PMON Sensor Monitoring Enhancement #
+
+
+### Revision 1.0 
+
+## Table of Content 
+1. [Scope](#Scope)
+2. [Definitions](#Definitions/Abbreviations)
+3. [Overview](#Overview)
+3. [Requirements](#Requirements)
+4. [High Level Design](#High-Level-Design)
+5. [CLI](#CLI-Enhancements)
+7. [Test](#Testing-Considerations)
+
+
+
+### Scope  
+
+This document covers the support for monitoring voltage and current sensor devices in SONiC. 
+
+### Definitions/Abbreviations 
+
+PMON - Platform Monitor container in SONiC.
+
+PSU - Power Supply Unit
+
+Voltage Sensor - Sensor device which can report a voltage measurement in the system.
+
+Current Sensor - Sensor device which can report current measurement in the system.
+
+Altitude Sensor - Sensor device which can report the altitude of the system.
+
+
+### Overview 
+
+Modern hardware systems have many different types of sensors and control devices. Voltage sensor devices can measure and in some cases control the voltages on the boards. Current sensor devices can measure current. It is also possible to have other types of sensors such as Altitude Sensors etc. These devices can report measurements from different parts of the system which are useful for monitoring system health. For example, voltage controller devices distribute power across different parts of the system such as  motherboard, daughterboards etc. and can report voltage measurements from there. Often these devices can report under-voltage/over-voltage faults which should be monitored to alert the operator about any power related failures in the system. This document provides an overview for monitoring voltage and current sensors in SONiC. The solution proposed in this document can be enhanced for other types of sensors as well.
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+
+### Requirements
+
+This HLD covers
+
+* Discovery of voltage and current sensor devices in the system
+* Monitoring the sensor devices periodically and update that data in Redis DB
+* Raise/Clear alarms if the sensor devices indicate readings which are unexpected and clear them if they return to a good state.
```

Comment:
@bmridul aren't these already done by `sensord`

Reply:
Answered in previous comment.